### ISSUE TITLE
[FR] Prodigy integration

### ISSUE THREAD

**Issue by strickvl** [voxel51/fiftyone#1871]
### Proposal Summary

Propose to add an integration with the popular annotation tool [Prodigy](https://prodi.gy), like what currently exists for LabelBox.

### Motivation

-   What is the use case for this feature?

Users want to iterate between their use of FiftyOne and labelling with ease. If they're using Prodigy, this process involves jumping through a few hoops currently.

-   Why is this use case valuable to support for FiftyOne users in general?

Prodigy is a popular tool. This feature will open up FiftyOne to users of Prodigy.

-   Why is this use case valuable to support for your project(s) or
    organization?

I use Prodigy.

-   Why is it currently difficult to achieve this use case? (please be as
    specific as possible about why related FiftyOne features and components are
    insufficient)

No integration currently exists.

### What areas of FiftyOne does this feature affect?

As far as I'm aware...

-   [x] App: FiftyOne application
-   [ ] Core: Core `fiftyone` Python library
-   [ ] Server: FiftyOne server

### Details

I will follow the suggestions made [here](https://voxel51.com/docs/fiftyone/user_guide/annotation.html#custom-annotation-backends) as well as the other examples where integrations currently exist.

### Willingness to contribute

The FiftyOne Community encourages new feature contributions. Would you or
another member of your organization be willing to contribute an implementation
of this feature?

-   [x] Yes. I can contribute this feature independently.
-   [x] Yes. I would be willing to contribute this feature with guidance from
        the FiftyOne community.
-   [ ] No. I cannot contribute this feature at this time.

I'll try implementing it myself, but I may need some help along the way!

**Comment by swheaton** [comment]
Wondering if you worked on this integration, need any pointers, etc.? Seeing no issue or PR activity in a long time

**Comment by strickvl** [comment]
@swheaton thanks for the ping on this! I haven't worked on it in a while, but Prodigy has released an update recently so I will get back into it! Are you a Prodigy user?

**Comment by swheaton** [comment]
Nope, voxel51 developer! I was trying to clean up some old GitHub issues and came across this. Also someone recently asked me at a conference if Fiftyone has a Prodigy integration
